lib/vector-core/src/config/proxy.rs

```diff
@@ -105,32 +107,36 @@ impl ProxyConfig {
         }
     }
 
-    fn http_intercept(&self) -> Intercept {
-        self.interceptor().intercept("http")
-    }
-
-    fn http_proxy(&self) -> Result<Option<Proxy>, InvalidUri> {
-        self.http
+    fn build_proxy(
+        &self,
+        proxy_scheme: &'static str,
+        proxy_url: &Option<String>,
+    ) -> Result<Option<Proxy>, InvalidUri> {
+        proxy_url
             .as_ref()
             .map(|url| {
-                url.parse()
-                    .map(|parsed| Proxy::new(self.http_intercept(), parsed))
+                url.parse().map(|parsed| {
+                    let mut proxy = Proxy::new(self.interceptor().intercept(proxy_scheme), parsed);
+                    if let Some(authority) = Url::parse(url).ok() {
```

Comment:
Is this the same thing?
```suggestion
                    if let Ok(authority) = Url::parse(url) {
```

Reply:
Whoops, fixed!